L1Trigger/L1THGCal/plugins/backend/HGCalSortingTruncationWrapper.cc

```diff
@@ -0,0 +1,134 @@
+                                         constituent.second->mipPt(),
+                                         imulticluster),
+            1.));
+
+      } else {
+        multiclusters_SA.at(imulticluster)
+            .addConstituent(l1thgcfirmware::HGCalCluster(constituent.second->centreProj().x(),
+                                                         constituent.second->centreProj().y(),
+                                                         constituent.second->centreProj().z(),
+                                                         triggerTools_.zside(constituent.second->detId()),
+                                                         triggerTools_.layerWithOffset(constituent.second->detId()),
+                                                         constituent.second->eta(),
+                                                         constituent.second->phi(),
+                                                         constituent.second->pt(),
+                                                         constituent.second->mipPt(),
+                                                         imulticluster),
+                            1.);
+      }
+      firstConstituent = false;
+      ++clusterIndex;
+    }
+  }
+}
+
+void HGCalSortingTruncationWrapper::convertAlgorithmOutputs(
+    const std::vector<l1thgcfirmware::HGCalMulticluster>& multiclusters_out,
+    const l1t::HGCalMulticlusterBxCollection& multiclusters_original,
+    l1t::HGCalMulticlusterBxCollection& multiclustersBXCollection) const {
+  std::vector<l1t::HGCalMulticluster> multiclusters;
+  multiclusters.reserve(multiclusters_out.size());
+  for (unsigned int imulticluster = 0; imulticluster < multiclusters_out.size(); ++imulticluster) {
+    unsigned multicluster_id = multiclusters_out[imulticluster].constituents().at(0).index_cmssw();
+    multiclusters.emplace_back(multiclusters_original[multicluster_id]);
+  }
+
+  for (const auto& multicluster : multiclusters) {
+    multiclustersBXCollection.push_back(0, multicluster);
+  }
+}
+
+void HGCalSortingTruncationWrapper::process(const l1t::HGCalMulticlusterBxCollection& inputMulticlusters,
+                                            l1t::HGCalMulticlusterBxCollection& outputMulticlusters) const {
+  l1thgcfirmware::HGCalMulticlusterSACollection multiclusters_SA;
+  convertCMSSWInputs(inputMulticlusters, multiclusters_SA);
+
+  l1thgcfirmware::HGCalMulticlusterSACollection multiclusters_finalized_SA;
+  sortAndTruncate(multiclusters_SA, multiclusters_finalized_SA);
+
+  convertAlgorithmOutputs(multiclusters_finalized_SA, inputMulticlusters, outputMulticlusters);
+}
+
+void HGCalSortingTruncationWrapper::sortAndTruncate(
+    const l1thgcfirmware::HGCalMulticlusterSACollection& inputMulticlusters,
+    l1thgcfirmware::HGCalMulticlusterSACollection& outputMulticlusters) const {
+  // Call SA sorting
+  theAlgo_.sortAndTruncate_SA(inputMulticlusters, outputMulticlusters, theConfiguration_);
+}
+
```

Comment:
Is it necessary to encapsulate a single line (`theAlgo_.sortAndTruncate_SA()`) into a wrapper function? Couldn't it be called directly inside `process()`?

Reply:
Not really, so I've implemented your suggestion.